### PR TITLE
support config replicaCount

### PR DESCRIPTION
**Pull request by Frapschen**

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
@Frapschen Thank you for the PR. Could you add check if `leaderElection` is enabled in case of `replicaCount` > `1` and fail otherwise?

**Comment by Frapschen** [comment]
@sumo-drosiek Is there a release plan？

**Comment by sumo-drosiek** [comment]
> @sumo-drosiek Is there a release plan？

We can release after the merge

**Comment by sumo-drosiek** [comment]
> @Frapschen Thank you for the PR. Could you add check if `leaderElection` is enabled in case of `replicaCount` > `1` and fail otherwise?

I was thinking more like about approach we have in [Sumo Logic Kubernetes Collection](https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/ca67ae6a29f3385a86c3bb23d8683952f5a3ac86/deploy/helm/sumologic/templates/checks.txt)

please see the following example:
```
{{/* Check if metrics provider is set to either fluentd or otelcol */}}
{{- if not (or (eq .Values.sumologic.metrics.metadata.provider "fluentd") (eq .Values.sumologic.metrics.metadata.provider "otelcol")) -}}
{{- fail "\nsumologic.metrics.metadata.provider should be set to either fluentd or otelcol" -}}
{{- end -}}
```

File: helm/tailing-sidecar-operator/templates/resources.yaml [hunk 1]
```diff
@@ -418,7 +418,11 @@ metadata:
   name: {{ include "tailing-sidecar-operator.fullname" . }}
   namespace: {{ .Release.Namespace }}
 spec:
-  replicas: 1
+  {{/* Check if leader election is set to true when replicaCount > 1 */}}
+  {{- if and (gt (.Values.operator.replicaCount | toString) "1") (not .Values.operator.leaderElection.enabled) }}
+  {{- fail "\nValues.operator.leaderElection should be set to true when replicaCount > 1" -}}
+  {{- end }}
```
Comment (sumo-drosiek) [inline helm/tailing-sidecar-operator/templates/resources.yaml:424]: Could you move it to `checks.txt`?

Comment (Frapschen) [reply helm/tailing-sidecar-operator/templates/resources.yaml]: done

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**